pydm/widgets/related_display_button.py

```diff
@@ -316,7 +343,25 @@ def context_menu(self):
             menu = QMenu(self)
         if len(menu.findChildren(QAction)) > 0:
             menu.addSeparator()
-        menu.addAction(self.open_in_new_window_action)
+        if len(self.filenames) <= 1:
+            action = menu.addAction(self.open_in_new_window_action)
+            return menu
+        sub_menu = menu.addMenu("Open in New Window")
+        for i, filename in enumerate(self.filenames):
```

Comment:
There is some repeating code here that I am not too happy about (from `_rebuild_menu`), but I did not want to change too much of existing code... should I refactor this a bit, any suggestions?

Reply:
The repetitive code seems to be due to the underlying separate storage of filenames/titles/macros, when it seems like they should all be bundled together (e.g., each entry might be a separate dictionary or `namedtuple` which would hold the filename, title, macros, etc.).

A simple first-pass that wouldn't require refactoring other stuff might be to add in a helper method that aggregates things for you:

```python
def _get_items(self):
   for i, filename in enumerate(self.filenames):
       item = {'filename': filename}
       if i >= len(self.titles):
          item['title'] = filename
       ... # fill in other stuff
       yield item
```

Reply:
Thanks @klauer for the suggestion. I added the helper method in 749937c.